### PR TITLE
Cut 1.5.0

### PR DESCRIPTION
**Pull request by brancz**
After an additional testing period of 30 days, there were no additional bugs found or features introduced. Due to no bugs being reported over an in total 41 days period, I feel no more pre-releases are necessary for a stable release.

This release's focus was a large architectural change in order to improve performance and resource usage of kube-state-metrics drastically. Special thanks to @mxinden for his hard work on this! See the changelog of the pre-releases for more detailed information and related pull requests.

@metalmatze @s-urbaniak @mxinden @squat @andyxning @zouyee 

@loburm @piosz @kawych @DirectXMan12 please stand by to publish the gcr.io image.

### CODE REVIEW

**Comment by brancz** [comment]
Added the additional `[CHANGE]` as a result of https://github.com/kubernetes/kube-state-metrics/issues/623. Typically we don't add non "feature" related internal things, but I feel it's justified in this case.

**Comment by ankon** [comment]
For #623: There is also a bit of code change in order I think:

```diff
diff --git a/README.md b/README.md
index ca6be8c..0fca2b6 100644
--- a/README.md
+++ b/README.md
@@ -102,12 +102,7 @@ additional metrics!
 See the [`Documentation`](Documentation) directory for more informations of the exposed metrics.
 
 ### Kube-state-metrics self metrics
-kube-state-metrics exposes its own metrics under `--telemetry-host` and `--telemetry-port` (default 81).
-
-| Metric name | Metric type | Description | Labels/tags |
-| ----------- | ----------- | ----------- | ----------- |
-| ksm_scrape_error_total   | Counter | Total scrape errors encountered when scraping a resource | `resource`=&lt;resource name&gt; |
-| ksm_resources_per_scrape | Summary | Number of resources returned per scrape | `resource`=&lt;resource name&gt; |
+kube-state-metrics exposes its own general process metrics under `--telemetry-host` and `--telemetry-port` (default 81).
 
 ### Resource recommendation
 
diff --git a/pkg/collectors/utils.go b/pkg/collectors/utils.go
index c15adc5..2828707 100644
--- a/pkg/collectors/utils.go
+++ b/pkg/collectors/utils.go
@@ -20,7 +20,6 @@ import (
 	"regexp"
 	"time"
 
-	"github.com/prometheus/client_golang/prometheus"
 	"k8s.io/api/core/v1"
 
 	"k8s.io/kube-state-metrics/pkg/metrics"
@@ -29,22 +28,6 @@ import (
 var (
 	resyncPeriod = 5 * time.Minute
 
-	ScrapeErrorTotalMetric = prometheus.NewCounterVec(
-		prometheus.CounterOpts{
-			Name: "ksm_scrape_error_total",
-			Help: "Total scrape errors encountered when scraping a resource",
-		},
-		[]string{"resource"},
-	)
-
-	ResourcesPerScrapeMetric = prometheus.NewSummaryVec(
-		prometheus.SummaryOpts{
-			Name: "ksm_resources_per_scrape",
-			Help: "Number of resources returned per scrape",
-		},
-		[]string{"resource"},
-	)
-
 	invalidLabelCharRE = regexp.MustCompile(`[^a-zA-Z0-9_]`)
 )
 
```

(preparing a PR, but my go-world is slow, so this is "untested" :D)

**Comment by brancz** [comment]
@ankon can we do only the docs changes for now, I'd prefer if the binary could stay the same from the last two pre-releases.

**Comment by ankon** [comment]
Sure!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
+## v1.5.0 / 2018-01-10
+
+After a testing period of 30 days, there were no additional bugs found or features introduced. Due to no bugs being reported over an in total 41 days, we feel no more pre-releases are necessary for a stable release.
+
+This release's focus was a large architectural change in order to improve performance and resource usage of kube-state-metrics drastically. Special thanks to @mxinden for his hard work on this! See the changelog of the pre-releases for more detailed information and related pull requests.
+
+An additional change has been requested to be listed in the release notes:
+
+* [CHANGE] Due to removal of the surrounding mechanism the `ksm_resources_per_scrape` metric no longer exists.
```
Comment (ankon) [inline CHANGELOG.md]: Small nit-pick: Both `ksm_resources_per_scrape` and `ksm_scrape_error_total` metrics actually.

Comment (brancz) [reply CHANGELOG.md]: You're absolutely right! Adapted.

**Comment by andyxning** [comment]
Again, i am curious about the reason to remove the kube-state-metrics telemetry metrics.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,13 @@
+## v1.5.0 / 2018-01-10
```
Comment (andyxning) [inline CHANGELOG.md]: This is actually 2019-01-10. :)

**Comment by brancz** [comment]
@andyxning as mentioned in the issue, the metrics don't exist anymore because objects are not being processed anymore during scrapes, so the metric has no relevance anymore as the mechanism completely changed.

File: CHANGELOG.md (hunk 2 above)
Comment (brancz) [reply CHANGELOG.md]: oops, good catch! :slightly_smiling_face:

**Comment by brancz** [comment]
Image published at `quay.io/coreos/kube-state-metrics:v1.5.0`

Please publish the gcr.io image @loburm @piosz @kawych @DirectXMan12

**Pull request merged**